OpenRA.Mods.Cnc/FileFormats/BlowfishKeyProvider.cs

```diff
@@ -90,9 +90,8 @@ static void KeyToBigNum(uint[] n, byte[] key, uint len)
 
 		static uint LenBigNum(uint[] n, uint len)
 		{
-			uint i;
-			i = len - 1;
-			while ((i >= 0) && (n[i] == 0)) i--;
+			var i = len - 1;
+			while (n[i] == 0) i--;
```

Comment:
NOTE: I don't know why my comment for this line was not added to my last review; I probably forgot to add it to the review. That is why I have been waiting.

`while (n[i] == 0) i--;` throws an out of range exception if `n` is zeros through `n[len]`. If that is guaranteed to never happen, then I agree that `while (n[i] == 0) i--;` would be faster.
```suggestion
while ((i >= 0) && (n[i] == 0)) i--;
```

Reply:
>  if `n` is zeros `n[len]` through `n[0]`

I'm not sure I understand what you mean there, but it's guaranteed that `i >= 0` because i is an `uint`.
Plus it's detected automatically as a  `'always true' expression`, otherwise I wouldn't dare to touch it.

That's why after the refactoring I decided to keep the explicit `uint` declaration (your nit above), since it helps understanding why `i` can never be `< 0` by design.

Reply:
Decrementing a `uint` when it is zero causes it to wrap around to the max value. I have this test code right here:
```c#
var someArray = new uint[] { 0, 0, 8, 1, 0, 0 };
var i = 2U - 1;
while (someArray[i] == 0) i--;
```

Reply:
This is actually a bug here! Try:
```c#
var someArray = new uint[] { 0, 0, 8, 1, 0, 0 };
var i = 2U - 1;
while (someArray[i] == 0) i--;
```
I'm removing the fixup request since fixing this bug is out of scope.